spring-kafka/src/main/java/org/springframework/kafka/listener/AbstractMessageListenerContainer.java

```diff
@@ -239,6 +256,123 @@ public void setAutoStartup(boolean autoStartup) {
 		this.autoStartup = autoStartup;
 	}
 
+	public ErrorHandler getErrorHandler() {
+		return this.errorHandler;
+	}
+
+	public void setErrorHandler(ErrorHandler errorHandler) {
+		this.errorHandler = errorHandler;
+	}
+
+	protected Executor getConsumerTaskExecutor() {
+		return this.consumerTaskExecutor;
+	}
+
+	/**
+	 * Set the executor for threads that poll the consumer.
+	 * @param consumerTaskExecutor the executor
+	 */
+	public void setConsumerTaskExecutor(Executor consumerTaskExecutor) {
+		this.consumerTaskExecutor = consumerTaskExecutor;
+	}
+
+	protected Executor getListenerTaskExecutor() {
+		return this.listenerTaskExecutor;
+	}
+
+	/**
+	 * Set the executor for threads that invoke the listener.
+	 * @param listenerTaskExecutor the executor.
+	 */
+	public void setListenerTaskExecutor(Executor listenerTaskExecutor) {
+		this.listenerTaskExecutor = listenerTaskExecutor;
+	}
+
+	protected long getPauseAfter() {
+		return this.pauseAfter;
+	}
+
+	/**
+	 * When using Kafka group management and {@link #setPauseEnabled(boolean)} is
+	 * true, the delay after which the consumer should be paused. Default 10000.
+	 * @param pauseAfter the delay.
+	 */
+	public void setPauseAfter(long pauseAfter) {
+		this.pauseAfter = pauseAfter;
+	}
+
+	protected boolean isPauseEnabled() {
+		return this.pauseEnabled;
+	}
+
+	/**
+	 * Set to true to avoid rebalancing when this consumer is slow or throws a
+	 * qualifying exception - pause the consumer.
+	 * Default: true.
+	 * @param pauseEnabled true to pause.
+	 * @see #setPauseAfter(long)
+	 */
+	public void setPauseEnabled(boolean pauseEnabled) {
+		this.pauseEnabled = pauseEnabled;
+	}
+
+	protected RetryTemplate getRetryTemplate() {
+		return this.retryTemplate;
+	}
+
+	/**
+	 * Set a retry template to retry deliveries.
+	 * @param retryTemplate the retry template.
+	 */
+	public void setRetryTemplate(RetryTemplate retryTemplate) {
+		this.retryTemplate = retryTemplate;
+	}
+
+	protected RecoveryCallback<Void> getRecoveryCallback() {
+		return this.recoveryCallback != null ? this.recoveryCallback :
+			new RecoveryCallback<Void>() {
```

Comment:
Looks like this instance can be done only once on the property directly.
I know @mbogoevici prefers private specific classes on the matter :smile: